### PR TITLE
wasi(windows): fixes readdir on written-after-opened directories 

### PR DESCRIPTION
**Pull request by mathetake**
extracted from #1057

### CODE REVIEW

**Comment by mathetake** [comment]
why ubuntu is also failing...

File: internal/sysfs/syscall_windows.go [hunk 1]
```diff
@@ -103,24 +102,42 @@ func rename(old, new string) (err error) {
...
+	path               string
+	flag               int
+	perm               fs.FileMode
+	readDirInitialized bool
+}
+
+// ReadDir implements fs.ReadDirFile.
+func (w *windowsWrappedFile) ReadDir(n int) ([]fs.DirEntry, error) {
+	if !w.readDirInitialized {
+		// On Windows, once the directory is opened, changes to the directory
```
Comment (codefromthecrypt) [inline internal/sysfs/syscall_windows.go:121]: do you have any link somewhere to cite this?

File: imports/wasi_snapshot_preview1/fs_test.go [hunk 2]
```diff
@@ -3879,3 +3879,50 @@ func requireOpenFile(t *testing.T, tmpDir string, pathName string, data []byte,
 
 	return mod, fd, log, r
 }
+
+// Test_fdReaddir_opened_file_written ensures that writing files to the already-opened directory
+// is visible. This is significant on Windows.
+// https://github.com/ziglang/zig/blob/2ccff5115454bab4898bae3de88f5619310bc5c1/lib/std/fs/test.zig#L156-L184
+func Test_fdReaddir_opened_file_written(t *testing.T) {
```
Comment (codefromthecrypt) [inline imports/wasi_snapshot_preview1/fs_test.go:3886]: can you add this test to dirfs_test or move it? usually I just convert it. that way we get it covered when implementing the next wasi. I don't think this is wasi-specific, even if it was noticed in wasi.

**Review by codefromthecrypt — APPROVED** [review]
nicely done!

**Pull request merged**